src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java

```diff
@@ -93,8 +112,21 @@ protected Executor tryResolve() throws Exception {
                 Future<Queue.Executable> future = item.getFuture().getStartCondition();
 
                 if (!future.isDone()) {
-                    // TODO JENKINS-26130 we might be able to detect that the item is blocked on an agent which has been deleted (not just offline), and abort ourselves
-                    LOGGER.log(Level.FINER, "{0} not yet started", item);
+                    Queue.Task task = item.task;
+                    Label nodeNameLabel = task.getAssignedLabel();  // Once execution started, this was set to the Computer name
+                    // See javadocs for ExecutorStepExecution.PlaceholderTask.label
+                    long waitTime = System.currentTimeMillis()-startTimeMillis;
+                    // TODO see if we need to do something with the ExecutorStepExecution.PlaceholderTask.runningTasks
+                    if (nodeNameLabel != null && task.getLastBuiltOn() == null) {  // Can't find node it was built on!
```

Comment:
This is a roundabout way of checking things. `getAssignedLabel()` may or may not be null prior to the start of the body: depends on whether a label was passed to the `node` step. And the same is true of `getLastBuiltOn()`: if a self-label was passed as an explicit argument, and the node is still around, it will be non-null even if the task has not yet been scheduled to an executor. This logic may be right but it is hard for me to say for sure—think it out carefully.

BTW if you need to know for sure whether the body has started, you can check `PlaceholderTask.cookie`.